### DISCUSSION TITLE
How to disable dressing.nvim in nvim-tree.lua?

### DISCUSSION THREAD

**Discussion by linrongbin16** [stevearc/dressing.nvim#81] [Q&A] [answered]
Hi, I'm using dressing.nvim and it looks great for me.

Only issue is that, when create/rename a file in nvim-tree.lua. Since my nvim-tree.lua is on left side, width=40. The input UI box is very narrow.

<img width="1920" alt="image" src="https://user-images.githubusercontent.com/6496887/213065355-fcf79691-d379-439b-967a-ea6db8574f97.png">

Is there any configuration can solve this issue? For example, fallback to native command-line when window width < 50?
How to disable dressing.nvim for filetype 'NvimTree'?

**Comment by stevearc** [comment, marked as answer]
The most straightforward change would be to open the input window relative to the editor instead of the window
```lua
require("dressing").setup({
  input = {
    relative = "editor",
  }
})
```
If you want to keep the current behavior and only make the input window relative to the editor when you're in a narrow window, try this:
```lua
require("dressing").setup({
  input = {
    get_config = function()
      if vim.api.nvim_win_get_width(0) < 50 then
        return {
          relative = "editor",
        }
      end
    end,
  }
})
```

**Reply by linrongbin16** [reply]
Thanks, I use your second option, issue solved!

**Reply by linrongbin16** [reply]
Hi, @stevearc ,

The default config has better position (but the width is a little bit small):
<img width="1920" alt="image" src="https://user-images.githubusercontent.com/6496887/215308399-8b738ed7-57cb-46c9-94b0-8f9e6e923d84.png">

If I set relative='editor', the input UI box position is center of the whole editor (the position is not precise):
<img width="1920" alt="image" src="https://user-images.githubusercontent.com/6496887/215308448-e253fadb-9c96-4611-876e-240e036509db.png">

I'm thinking, if there's a way, position is near the cursor, and width is bigger....

**Reply by linrongbin16** [reply]
I finally use the `override` function to do this:

```
local function OverrideConfig(conf)
    if vim.api.nvim_win_get_width(0) <= 40 then
        conf.width = 50
    end
    return conf
end

require("dressing").setup({
    input = {
        override = OverrideConfig,
    },
})
```

This will detect if window width <= 40, if yes, it will forcely set dressing input UI box width to 50.

It's much longer than default UI box:

<img width="1920" alt="image" src="https://user-images.githubusercontent.com/6496887/215313585-5cdab234-3acf-498c-b5a2-b7eabcb601ab.png">
